### PR TITLE
Add ProhibitOverlap option to PeriodicConfig

### PR DESCRIPTION
**Pull request by dadgar**
This PR adds the ability to disallow overlapping periodic jobs.

### CODE REVIEW

**Comment by c4milo** [comment]
nice work @dadgar. I would probably add an additional log message for when users have a `count` greater than 1 as well as the `prohibit_overlap = true`. Just to let them know the extra instances are not going to be launched because of that.

File: nomad/state/schema.go [hunk 1]
```diff
@@ -118,6 +118,15 @@ func jobTableSchema() *memdb.TableSchema {
 					Conditional: jobIsPeriodic,
 				},
 			},
+			"parent": &memdb.IndexSchema{
```
Comment (armon) [inline nomad/state/schema.go]: Can we instead just do an "id_prefix" lookup and avoid the additional index?

File: nomad/state/state_store.go [hunk 2]
```diff
@@ -419,6 +419,18 @@ func (s *StateStore) JobsByGC(gc bool) (memdb.ResultIterator, error) {
 	return iter, nil
 }
 
+// ChildJobs returns an iterator over all the children of the passed job.
+func (s *StateStore) ChildJobs(id string) (memdb.ResultIterator, error) {
+	txn := s.db.Txn(false)
+
+	// Scan all jobs whose parent is the passed id.
+	iter, err := txn.Get("jobs", "parent", id)
```
Comment (armon) [inline nomad/state/state_store.go]: If we do a loop over "id_prefix" and check if "parent == id" the scan cost is the same but avoids another index

**Comment by dadgar** [comment]
@c4milo: count > 1 will still work. This prohibits overlapping jobs not tasks

**Pull request merged**